### DISCUSSION TITLE
How do you implment ECH, ENSI and local DOH using a Lets Encrypt cert?

### DISCUSSION THREAD

**Discussion by eshankiyer** [DNSCrypt/dnscrypt-proxy#2098] [Q&A]
^^^

I've tried pretty much every setting with certbot and was unable to generate a LetsEncrypt certificate for my local DOH server.

**Comment by jedisct1** [comment]
You can try [acme.sh](https://github.com/acmesh-official/acme.sh) which is fairly ease to use.
